### PR TITLE
 [ADF-3671] Add automation tests for Form Widgets

### PR DESCRIPTION
**Pull request by davidcanonieto**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe: E2e tests

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-3617
https://issues.alfresco.com/jira/browse/ADF-3671

**What is the new behaviour?**
Added automation tests for form widgets.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-3617
https://issues.alfresco.com/jira/browse/ADF-3671

### CODE REVIEW

File: e2e/pages/adf/process_services/widgets/checkbox.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FormFields = require('../formFields');
+import Util = require('../../../../util/util');
+import { by, element } from 'protractor';
+
+export class Checkbox {
```
Comment (cristinaj) [inline e2e/pages/adf/process_services/widgets/checkbox.ts]: Some of the widgets classes contain Widget in the name, like TextWidget, and others don't. I suggest all to be the same, either contain it, either don't.

File: e2e/models/ACS/filesModel.js [hunk 2]
```diff
@@ -15,20 +15,27 @@
  * limitations under the License.
  */
 
-var FileModel = require('./fileModel');
+var FormFields = require('../formFields');
```
Comment (cristinaj) [inline e2e/models/ACS/filesModel.js]: Any reason why fileModel.js file contains DisplayText var?

File: e2e/process-services/widgets/date_time_widget.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    });
+
+    afterAll(async (done) => {
+        await alfrescoJsApi.activiti.processApi.deleteProcessInstance(process.id);
+        await alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+        await alfrescoJsApi.activiti.adminTenantsApi.deleteTenant(processUserModel.tenantId);
+        done();
+    });
+
+    it('C268818] Date and time widget - General properties', () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/date_time_widget.e2e.ts]: This line is missing a '['

File: e2e/process-services/widgets/multi_line_widget.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    it('[C268184] Multi-line Text Widget - Advanced Properties - Regex Pattern property', async () => {
+        widget.multilineTextWidget().setValue(app.FIELD.multiSimple, 'TEST');
+        widget.multilineTextWidget().setValue(app.FIELD.multiRegexp, '3');
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeTruthy();
+        expect(widget.multilineTextWidget().getErrorMessage(app.FIELD.multiRegexp)).toBe('Enter a different value');
+        widget.multilineTextWidget().setValue(app.FIELD.multiRegexp, 'TE');
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeFalsy();
+    });
+
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/multi_line_widget.e2e.ts]: Could you merge the 2 it above into one and name it : Should be able to set advanced properties for Multi-line Text Widget ?

File: e2e/process-services/widgets/multi_line_widget.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    it('[C268232] Multi-line Text Widget - Visibility properties', async () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/multi_line_widget.e2e.ts]: Could you change the name to : Should be able to set visibility properties for Multi-line Text Widget ?

File: e2e/process-services/widgets/number_widget.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    });
+
+    afterAll(async (done) => {
+        await alfrescoJsApi.activiti.processApi.deleteProcessInstance(process.id);
+        await alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+        await alfrescoJsApi.activiti.adminTenantsApi.deleteTenant(processUserModel.tenantId);
+        done();
+    });
+
+    it('[C269111] Number Widget - General Properties', () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/number_widget.e2e.ts]: Could you change the name to : Should be able to set general properties for Number Widget ?

File: e2e/process-services/widgets/number_widget.e2e.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeTruthy();
+        expect(widget.numberWidget().getNumberFieldLabel(app.FIELD.number_general)).toContain('Number General');
+        expect(widget.numberWidget().getPlaceholder(app.FIELD.number_general)).toContain('Type a number');
+
+        widget.numberWidget().setFieldValue(app.FIELD.number_general, 2);
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeFalsy();
+    });
+
+    it('[C274702] Number Widget - Advanced and visibility properties', () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/number_widget.e2e.ts]: Could you change the name to : Should be able to set advanced and visibility properties for Number Widget ?

File: e2e/process-services/widgets/people_widget.e2e.ts [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    });
+
+    afterAll(async (done) => {
+        await alfrescoJsApi.activiti.processApi.deleteProcessInstance(process.id);
+        await alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+        await alfrescoJsApi.activiti.adminTenantsApi.deleteTenant(processUserModel.tenantId);
+        done();
+    });
+
+    it('[C212870] should check People widget', () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/people_widget.e2e.ts]: The test case ID is wrong

File: e2e/process-services/widgets/people_widget.e2e.ts [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        taskPage.formFields().checkWidgetIsVisible(app.FIELD.widget_id);
+
+        let admin = processUserModel.firstName + ' ' + processUserModel.lastName;
+        widget.peopleWidget().insertUser(app.FIELD.widget_id, admin.charAt(0));
+        widget.peopleWidget().checkDropDownListIsDisplayed();
+        widget.peopleWidget().checkUserIsListed(admin);
+        widget.peopleWidget().selectUserFromDropDown(admin);
+    });
+
+    it('[C274707] Add people widget - Visibility', () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/people_widget.e2e.ts]: Could you change the name to : Should be possible to set visibility properties for People Widget ?

File: e2e/process-services/widgets/radio_buttons_widget.e2e.ts [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    });
+
+    afterAll(async (done) => {
+        await alfrescoJsApi.activiti.processApi.deleteProcessInstance(process.id);
+        await alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+        await alfrescoJsApi.activiti.adminTenantsApi.deleteTenant(processUserModel.tenantId);
+        done();
+    });
+
+    it('[C277316] Radio buttons widget - default behaviour', () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/radio_buttons_widget.e2e.ts]: Could you change the name to : Should display empty radio buttons when no preselection is configured ?

File: e2e/process-services/widgets/radio_buttons_widget.e2e.ts [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        widget.checkboxWidget().clickCheckboxInput(app.FIELD.checkbox_id);
+        widget.radioWidget().isSelectionClean(app.FIELD.radio_buttons_id);
+    });
+
+    it('[C274704] Radio buttons widget - Visibility', () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/radio_buttons_widget.e2e.ts]: Could you change the name to : Should be able to set visibility properties for Radio Button widget ?

File: e2e/process-services/widgets/text_widget.e2e.ts [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+    });
+
+    afterAll(async (done) => {
+        await alfrescoJsApi.activiti.processApi.deleteProcessInstance(process.id);
+        await alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+        await alfrescoJsApi.activiti.adminTenantsApi.deleteTenant(processUserModel.tenantId);
+        done();
+    });
+
+    it('[C268157] General Properties', async () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/text_widget.e2e.ts]: Could you change the name to : Should be able to set general properties for Text widget

File: e2e/process-services/widgets/text_widget.e2e.ts [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+        let label = widget.textWidget().getFieldLabel(app.FIELD.simpleText);
+        expect(label).toBe('textSimple*');
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeTruthy();
+        let placeHolder = widget.textWidget().getFieldPlaceHolder(app.FIELD.simpleText);
+        expect(placeHolder).toBe('Type something...');
+        widget.textWidget().setValue(app.FIELD.simpleText, 'TEST');
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeFalsy();
+    });
+
+    it('[C268170] Min-max length properties', async () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/text_widget.e2e.ts]: Could you change the name to : Should be possible to set min-max length properties for Text widget ?

File: e2e/process-services/widgets/text_widget.e2e.ts [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+        widget.textWidget().setValue(app.FIELD.textMinMax, 'A');
+        expect(widget.textWidget().getErrorMessage(app.FIELD.textMinMax)).toBe('Enter at least 4 characters');
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeTruthy();
+        widget.textWidget().setValue(app.FIELD.textMinMax, 'AAAAAAAAAAA');
+        expect(widget.textWidget().getErrorMessage(app.FIELD.textMinMax)).toBe('Enter no more than 10 characters');
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeTruthy();
+    });
+
+    it('[C268171] Input mask reversed checkbox properties', async () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/text_widget.e2e.ts]: Could you change the name to : Should be able to set input mask reversed checkbox properties for Text widget ?

File: e2e/process-services/widgets/text_widget.e2e.ts [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+        widget.textWidget().setValue(app.FIELD.textMask, '18951523');
+        expect(widget.textWidget().getFieldValue(app.FIELD.textMask)).toBe('1895-1523');
+    });
+
+    it('[C268171] Input mask reversed checkbox properties', async () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/text_widget.e2e.ts]: Can you merge this it to the one above ?

File: e2e/process-services/widgets/text_widget.e2e.ts [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+        widget.textWidget().setValue(app.FIELD.textMaskReversed, '1234567899');
+        expect(widget.textWidget().getFieldValue(app.FIELD.textMaskReversed)).toBe('3456-7899');
+    });
+
+    it('[C268177] Regex Pattern property', async () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/text_widget.e2e.ts]: Could you change the name to : Should be able to set Regex Pattern property for Text widget ?

File: e2e/process-services/widgets/text_widget.e2e.ts [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+        widget.textWidget().setValue(app.FIELD.simpleText, 'TEST');
+        widget.textWidget().setValue(app.FIELD.textRegexp, 'T');
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeTruthy();
+        expect(widget.textWidget().getErrorMessage(app.FIELD.textRegexp)).toBe('Enter a different value');
+        widget.textWidget().setValue(app.FIELD.textRegexp, 'TE');
+        expect(taskPage.formFields().isCompleteFormButtonDisabled()).toBeFalsy();
+    });
+
+    it('[C274712] Visibility condition', async () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/text_widget.e2e.ts]: Could you change the name to : Should be able to set visibility properties for Text widget ?

File: e2e/util/resources.js [hunk 18]
```diff
@@ -112,6 +112,207 @@ exports.Files = {
...
+
+        DYNAMIC_TABLE: {
+            formName: "DynamicTableProcess",
+            title: "DynamicTableProcess",
+            processName: "DynamicTableProcess",
+            FIELD: {
+                dynamic_table_age_id: "label2",
+                dynamic_table_date_id: "label",
+                checkbox_id: "viewtable",
+                dataTime_input_id: "datetime",
```
Comment (marouanbentaleb) [inline e2e/util/resources.js]: Could you change dataTime_input_id to dateTime_input_id ?

File: e2e/util/resources.js [hunk 19]
```diff
@@ -112,6 +112,207 @@ exports.Files = {
...
+
+        DYNAMIC_TABLE_USERS: {
+            formName: "ALL_WIDGETS",
+            title: "ALL_WIDGETS",
+            processName: "ALL_WIDGETS",
+            FIELD: {
+                dynamic_table_age_id: "label2",
+                dynamic_table_date_id: "label",
+                checkbox_id: "viewtable",
+                dataTime_input_id: "datetime",
```
Comment (marouanbentaleb) [inline e2e/util/resources.js]: Could you change dataTime_input_id to dateTime_input_id ?

File: e2e/process-services/widgets/multi_line_widget.e2e.ts [hunk 20]
```diff
@@ -0,0 +1,112 @@
...
+    });
+
+    afterAll(async (done) => {
+        await alfrescoJsApi.activiti.processApi.deleteProcessInstance(process.id);
+        await alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+        await alfrescoJsApi.activiti.adminTenantsApi.deleteTenant(processUserModel.tenantId);
+        done();
+    });
+
+    it('[C268182] Multi-line Text Widget - General Properties', async () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/multi_line_widget.e2e.ts]: Could you remove async ? If needed use `browser.controlFlow().excute(async () => {})` inside the it

File: e2e/process-services/widgets/dynamic_table_widget.e2e.ts [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+            widget.dateTimeWidget().selectHour('8');
+            widget.dateTimeWidget().selectMinute('30');
+            widget.dateTimeWidget().clearDateTimeInput(app.FIELD.dataTime_input_id);
+
+            widget.dynamicTable().clickSaveButton();
+            widget.dynamicTable().getTableRow(0);
+        });
+    });
+
+    describe('with date widget', () => {
```
Comment (marouanbentaleb) [inline e2e/process-services/widgets/dynamic_table_widget.e2e.ts]: Don't forget to change this describe too

**Pull request merged**